content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,774 @@
+# delete the original pod
+kubectl -n verify-pod-security delete pod busybox-restricted
+
+# create the pod again with new runAsUser
+cat <<EOF | kubectl -n verify-pod-security apply -f -
+apiVersion: v1
+kind: Pod
+metadata:
+  name: busybox-restricted
+spec:
+  securityContext:
+    runAsUser: 65534
+  containers:
+  - name: busybox
+    image: busybox
+    args:
+    - sleep
+    - "1000000"
+    securityContext:
+      seccompProfile:
+        type: RuntimeDefault
+      runAsNonRoot: true
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+          - ALL
+        add:
+          - NET_BIND_SERVICE
+EOF
+```
+
+Run `kubectl -n verify-pod-security get pods` to verify it is running. The output is similar to this:
+
+```
+NAME                 READY   STATUS    RESTARTS   AGE
+busybox-restricted   1/1     Running   0          25s
+```
+
+Clean up the demo (restricted pod and namespace) with:
+
+```shell
+kubectl delete namespace verify-pod-security
+```
+
+At this point, if you wanted to dive deeper into linux permissions or what is permitted for a certain container, exec into the control plane and play around with `containerd` and `crictl inspect`.
+
+```shell
+# if using docker, shell into the control plane
+docker exec -it kind-control-plane bash
+
+# list running containers
+crictl ps
+
+# inspect each one by container ID
+crictl inspect <CONTAINER ID>
+```
+
+### Applying a cluster-wide policy
+
+In addition to applying labels to namespaces to configure policy you can also configure cluster-wide policies and exemptions using the AdmissionConfiguration resource. 
+
+Using this resource, policy definitions are applied cluster-wide by default and any policy that is applied via namespace labels will take precedence. 
+
+There is no runtime configurable API for this resource and a cluster administrator would need to specify a path to the file below via the `--admission-control-config-file` flag on the API server. 
+
+In the following resource we are enforcing the baseline policy and warning and auditing the baseline policy. We are also making the kube-system namespace exempt from this policy.
+
+It's not recommended to alter control plane / clusters after install, so let's build a new cluster with a default policy on all namespaces.
+
+First, delete the current cluster.
+
+```shell
+kind delete cluster
+```
+
+Create a Pod Security configuration. 
+
+<!-- TODO: link to the official docs after posted: https://github.com/kubernetes/website/blob/dev-1.23/content/en/docs/tasks/configure-pod-container/enforce-standards-admission-controller.md?plain=1#L25-L55
+--->
+
+```shell
+cat <<EOF > pod-security.yaml
+apiVersion: apiserver.config.k8s.io/v1
+kind: AdmissionConfiguration
+plugins:
+- name: PodSecurity
+  configuration:
+    apiVersion: pod-security.admission.config.k8s.io/v1beta1
+    kind: PodSecurityConfiguration
+    defaults:
+      enforce: "baseline"
+      enforce-version: "latest"
+      audit: "baseline"
```

Comment:
```suggestion
      audit: "restricted"
```

Reply:
I'm ok with this suggestion, but curious about your reasoning why? I don't care much if it's `baseline` or `restricted` for auditing, but I approached it as a cluster operator of, "I care historically what's allowed or rejected by PSA" vs. auditing a theoretical higher level of restrictions. I could see value in this approach though if you were interested in, "How many workloads do I allow today that would fail a more restrictive level?" but maybe I would use `warn` for that.

I see the modes like:
- `warn` == for developers / CD signal
- `audit` == for operators / compliance

No matter what approach is used, it might be worth talking about "why;" kind of similar to how we talk about when to `warn`. Wdyt?

Reply:
To be honest, I was only trying to align the label here with the text above and the tutorial for consistency :D 

In terms of personas though, you make a good point. I was mostly thinking about the latter when selecting these levels for audit i.e. "How many workloads do I allow today that would fail a more restrictive level?". But the other scenario is valid too. 

As long as we pick one consistently and share the rationale behind it prior to the `yaml`, Happy with either of them.

Reply:
I think it's worth adding a blurb about why the levels are chosen, similar to the the "when to warn?" section. I'll probably keep it as is with `baseline` though. Maybe you could do the same for using `restricted` in the tutorial? Open to alternative suggestions too!

Reply:
yes that works!